src/qibo/abstractions/circuit.py

```diff
@@ -955,7 +955,12 @@ def draw(self, line_wrap=70) -> str:
                   "crx": "RX", "cry": "RY", "crz": "RZ",
                   "cu1": "U1", "cu3": "U3", "ccx": "X",
                   "id": "I", "measure": "M", "fsim": "f",
-                  "generalizedfsim": "gf", "Unitary": "U", "fswap":"fx"}
+                  "generalizedfsim": "gf", "Unitary": "U", "fswap":"fx",
+                  "PauliNoiseChannel": "Pc", "KrausChannel": "Kc",
+                  "UnitaryChannel": "Uc", "ThermalRelaxationChannel": "Tc",
+                  "ResetChannel": "Rc", "PartialTrace": "Pt",
+                  "EntanglementEntropy": "EE", "Norm": "N",
+                  "Overlap": "O", "Energy": "E", "Gap": "G"}
```

Comment:
Do you think it would be useful to have a prefix (eg. Cb) for all callback gates, something like (CbEE, CbN, CbE, etc.) so that they can be distinguished are distinguishable from normal gates when calling `circuit.draw()`? 

Also, the gap callback is probably not needed here as this callback can only be used in an adiabatic evolution model, not in a circuit.

Reply:
Maybe prefix for the callbacks and suffix for the channels, just be consistent with the name of the gates, so for example the `PauliNoiseGate` should be something like `PNCh`.

Reply:
You mean adding like a legend that explains that `Ch` stands for channel and `Cb` for callback or something more complicated like a full legend of all the gates used in the circuit?

Reply:
Yes, for this and maybe all other gates, we could have an extra flag `legend` so the user can switch on/off.

Reply:
I coded small example where I removed the prefix and the suffix and I added a legend at the bottom.
Here are a few examples:
```python
from qibo.models import Circuit
from qibo import gates

c2 = Circuit(2, density_matrix=True)
c2.add(gates.H(0))
c2.add(gates.PauliNoiseChannel(0, 0.1, 0.0, 0.2))
c2.add(gates.H(1))
c2.add(gates.PauliNoiseChannel(1, 0.0, 0.2, 0.1))
c2.add(gates.CNOT(0, 1))
c2.add(gates.PauliNoiseChannel(0, 0.1, 0.0, 0.2))
c2.add(gates.PauliNoiseChannel(1, 0.0, 0.2, 0.1))
print(c2.draw(legend=True))
```
With output:
```
q0: ─H─PN─o─PN─
q1: ─H─PN─X─PN─

 Legend: 
| Gate              | Symbol   |
|-------------------+----------|
| PauliNoiseChannel | PN       |
```
Or 
```python
from qibo import models, gates, callbacks

# create entropy callback where qubit 0 is the first subsystem
entropy = callbacks.EntanglementEntropy([0])

# initialize circuit with 2 qubits and add gates
c = models.Circuit(2) # state is |00> (entropy = 0)
c.add(gates.CallbackGate(entropy)) # performs entropy calculation in the initial state
c.add(gates.H(0)) # state is |+0> (entropy = 0)
c.add(gates.CallbackGate(entropy)) # performs entropy calculation after H
c.add(gates.CNOT(0, 1)) # state is |00> + |11> (entropy = 1))
c.add(gates.CallbackGate(entropy)) # performs entropy calculation after CNOT

# execute the circuit using the callback
final_state = c()
print(c.draw(legend=True))
```
with output:
```
q0: ─EE─H─EE─o─EE─
q1: ─EE───EE─X─EE─

 Legend: 
| Gate                | Symbol   |
|---------------------+----------|
| EntanglementEntropy | EE       |
```
For the time being I think that we can add to the legend only the gates which don't have a standard notation, so the channels and the callbacks.
I made the table using [tabulate](https://pypi.org/project/tabulate/). Let me know if this approach could work.

Reply:
I like it, @stavros11 what do you think?

Reply:
Ok, maybe I'll add also a test with the legend.
I should also add the tabulate package as an installation requirement at this point, correct?

Reply:
> I should also add the tabulate package as an installation requirement at this point, correct?

Yes, I believe so, you need to add it to `install_requires` in setup.py. Would be good if you also add a test. You can add a `@pytest.mark.parametrize("legend", [False, True])` in the current test if you don't want to write a new one.

Reply:
Sure, this is exactly what I'm doing, thanks!